src/hooks/Circles/utils/optimisticallyDeletePosts.ts

```diff
@@ -34,49 +44,59 @@ export const optimisticallyDeletePosts = ({
           return undefined;
         }
 
-        removeRepliesById(currentData, postId);
-        return currentData;
+        const newData = cloneDeep(currentData);
+        deletePostById(newData.post.replies?.edges, postId);
+        return newData;
       });
     });
 };
 
-const removeEdgeById = (currentData: InfinitePostsData, id: string) =>
-  forEach(currentData?.pages, (page) => {
-    forEach(page.postsV2.edges, (edge) => {
-      const deleted = remove(
-        edge.node.replies?.edges,
-        (replyEdge) => replyEdge.node.id === id,
-      );
-      edge.node.replyCount -= deleted.length;
-    });
-    remove(page.postsV2.edges, (edge) => edge.node.id === id);
-  });
-
-const removeRepliesById = (
-  currentData: PostDetailsPostQueryResponse,
-  id: string,
-  depth: number = 7, // set some default max depth
+const deletePostById = (
+  posts: { node: Post }[],
+  postId: string,
+  maxDepth: number = 20,
+  depth: number = 0,
 ) => {
-  let count = 0;
-  const recursiveRemove = (post: Post, currentDepth: number) => {
-    if (currentDepth > depth) {
-      return;
-    }
+  // limit the depth of recursion
+  if (depth > maxDepth) {
+    return true;
```

Comment:
Should this be `false` so line 94 doesn't get a false positive?

Reply:
A true here simply signals that it is time to exit, not that we've succeeded or failed.